File: bigO/js/bigO.js
function sumArray(array) {
    let sum = 0;  // O(1) - Constant initialization
    for (let element of array) {  // O(n) - Iterates through all elements of the array
        sum += element;  // O(1) - Adds each element
    }
    return sum;  // O(1) - Returns the final result
}

// Analysis:
// - The loop is the dominant element with O(n).
// - Total complexity: O(n).

File: bigO/python/bigO.py
def sum_array(array):
    # Initialize the sum to 0
    sum = 0  # O(1) - Constant time operation
    for element in array:  # O(n) - Iterate through all elements of the array
        sum += element  # O(1) - Add each element to the sum
    return sum  # O(1) - Return the final result

# Analysis:
# - The loop is the dominant factor, with O(n).
# - Overall complexity: O(n).